sw/host/opentitanlib/src/transport/cw310/mod.rs

```diff
@@ -116,10 +118,20 @@ impl Transport for CW310 {
         Ok(Rc::clone(inner.spi.as_ref().unwrap()))
     }
 
-    fn fpga_program(&self, bitstream: &[u8]) -> Result<()> {
-        let usb = self.device.borrow();
-        usb.spi1_enable(false)?;
-        usb.pin_set_state(CW310::PIN_JTAG, true)?;
-        usb.fpga_program(bitstream)
+    fn dispatch(&self, action: &dyn Any) -> Result<Option<Box<dyn Serialize>>> {
+        if let Some(fpga_program) = action.downcast_ref::<FpgaProgram>() {
```

Comment:
This is up to you, but I prefer to write things in a way that minimizes indentation. I.e.,

```rust
let fpga_program = action.downcast_ref::<FpgaProgram>().ok_or(TransportError::...)?;
```

I would actually personally drop the turbofish and write `let fpga_program: &FpgaProgram = ...` but it's amatter of taste.